Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/TeamsController.cs

```diff
@@ -562,129 +564,238 @@ private async Task<List<ShiftsIntegResponse>> ProcessSwapShiftRequest(RequestMod
 
-                // Logging to ApplicationInsights the OpenShiftRequestId.
-                this.telemetryClient.TrackTrace("OpenShiftRequestId = " + finalOpenShiftRequest.Id);
+            // Update the open shift request to Approved in the ShiftStatus column.
+            await this.openShiftRequestMappingEntityProvider.SaveOrUpdateOpenShiftRequestMappingEntityAsync(openShiftRequestEntityToUpdate).ConfigureAwait(false);
 
-                // Find the open shift request for which we update the ShiftsStatus to Approved.
-                var openShiftRequestEntityToUpdate = await this.openShiftRequestMappingEntityProvider.GetOpenShiftRequestMappingEntityByOpenShiftIdAsync(
-                    finalOpenShift.Id,
-                    finalOpenShiftRequest.Id).ConfigureAwait(false);
+            // Delete the open shift entity accordingly from the OpenShiftEntityMapping table in Azure Table storage as the open shift request has been approved.
+            await this.openShiftMappingEntityProvider.DeleteOrphanDataFromOpenShiftMappingByOpenShiftIdAsync(openShift.Id).ConfigureAwait(false);
 
-                openShiftRequestEntityToUpdate.ShiftsStatus = finalOpenShiftRequest.State;
+            // Adding response for delete open shift.
+            responseModelList.Add(GenerateResponse(openShift.Id, HttpStatusCode.OK, null, null));
 
-                // Update the open shift request to Approved in the ShiftStatus column.
-                await this.openShiftRequestMappingEntityProvider.SaveOrUpdateOpenShiftRequestMappingEntityAsync(openShiftRequestEntityToUpdate).ConfigureAwait(false);
+            // Adding response for approved open shift request.
+            responseModelList.Add(GenerateResponse(openShiftRequest.Id, HttpStatusCode.OK, null, null));
 
-                // Delete the open shift entity accordingly from the OpenShiftEntityMapping table in Azure Table storage as the open shift request has been approved.
-                await this.openShiftMappingEntityProvider.DeleteOrphanDataFromOpenShiftMappingByOpenShiftIdAsync(finalOpenShift.Id).ConfigureAwait(false);
+            this.telemetryClient.TrackTrace($"Finished approving OpenShiftRequest {openShiftRequest.Id}");
+        }
 
-                // Adding response for delete open shift.
-                integrationResponse = GenerateResponse(finalOpenShift.Id, HttpStatusCode.OK, null, null);
-                responseModelList.Add(integrationResponse);
+        /// <summary>
+        /// Gets the temporary shift.
+        /// Creates a new shift using all of the temporary shift's data besides the RowKey.
+        /// Deletes the temporary shift.
+        /// </summary>
+        /// <param name="openShiftRequest">An open shift request.</param>
+        /// <param name="shift">A shift.</param>
+        /// <param name="kronosTimeZone">The Kronos time zone.</param>
+        /// <param name="responseModelList">The list of responses.</param>
+        private async Task DealWithTempShift(
```

Comment:
I'm sure you can come up with a better name for this e.g. CreateShiftFromTempShift or something?